drivers/comparator/comparator_stm32_comp.c

```diff
@@ -0,0 +1,324 @@
+			LL_COMP_Lock(comp);
+		}
+		LL_COMP_Enable(comp);
+	}
+
+#ifdef CONFIG_PM_DEVICE
+	if (action == PM_DEVICE_ACTION_SUSPEND) {
+		LL_COMP_Disable(comp);
+	}
+#endif
+
+	return 0;
+}
+
+static void stm32_comp_irq_handler(const struct device *dev)
+{
+	const struct stm32_comp_config *cfg = dev->config;
+	struct stm32_comp_data *data = dev->data;
+
+	if (stm32_exti_is_pending(cfg->exti_line_number)) {
+		stm32_exti_clear_pending(cfg->exti_line_number);
+	}
+
+	if (data->callback == NULL) {
+		return;
+	}
+
+	data->callback(dev, data->user_data);
+}
+
+static int stm32_comp_init(const struct device *dev)
+{
+	const struct device *const clk = DEVICE_DT_GET(STM32_CLOCK_CONTROL_NODE);
+	const struct stm32_comp_config *cfg = dev->config;
+	COMP_TypeDef *comp = cfg->comp;
+	int ret = 0;
+
+	if (!device_is_ready(clk)) {
+		LOG_ERR("%s clock control device not ready", dev->name);
+		return -ENODEV;
+	}
+
+	/* Enable COMP bus clock */
+	ret = clock_control_on(clk, &cfg->pclken[0]);
+	if (ret != 0) {
+		LOG_ERR("%s clock op failed (%d)", dev->name, ret);
+		return ret;
+	}
+
+	/* Enable COMP clock source */
+	clock_control_configure(clk, &cfg->pclken[1], NULL);
+	if (ret != 0) {
+		LOG_ERR("%s clock configure failed (%d)", dev->name, ret);
+		return ret;
+	}
+
+	/* Configure COMP inputs as specified in Device Tree, if any */
+	ret = pinctrl_apply_state(cfg->pincfg, PINCTRL_STATE_DEFAULT);
+	if (ret < 0 && ret != -ENOENT) {
+		/*
+		 * If the COMP is used only with internal channels, then no pinctrl is
+		 * provided in Device Tree, and pinctrl_apply_state returns -ENOENT,
+		 * but this should not be treated as an error.
+		 */
+		LOG_ERR("%s pinctrl setup failed (%d)", dev->name, ret);
+		return ret;
+	}
+
+	if (LL_COMP_IsLocked(comp)) {
+		/* COMP instance shall not be locked */
+		LOG_ERR("%s COMP instance is locked", dev->name);
+		return -EACCES;
+	}
+
+#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32h7_comp)
+	LL_COMP_SetPowerMode(comp, cfg->power_mode);
+#endif /* DT_HAS_COMPAT_STATUS_OKAY(st_stm32h7_comp) */
+
+	LL_COMP_SetInputMinus(comp, cfg->input_minus);
+	LL_COMP_SetInputPlus(comp, cfg->input_plus);
+	LL_COMP_SetInputHysteresis(comp, cfg->hysteresis);
+	LL_COMP_SetOutputPolarity(comp, cfg->invert_output);
+	LL_COMP_SetOutputBlankingSource(comp, cfg->blank_sel);
+
+	if (cfg->miller_effect_hold_enable) {
+#if defined(CONFIG_SOC_SERIES_STM32G4X)
+		SET_BIT(comp->CSR, BIT(1U));
+#endif
+	}
+
+	cfg->irq_init();
+
+	return pm_device_driver_init(dev, stm32_comp_pm_callback);
+}
+
+#define STM32_COMP_IRQ_HANDLER_SYM(inst)	_CONCAT(stm32_comp_irq_init, inst)
+
+#define STM32_COMP_IRQ_HANDLER_DEFINE(inst)				\
+	static void STM32_COMP_IRQ_HANDLER_SYM(inst)(void)		\
+	{							\
+		IRQ_CONNECT(DT_INST_IRQN(inst), DT_INST_IRQ(inst, priority), \
+			stm32_comp_irq_handler, DEVICE_DT_INST_GET(inst), 0); \
+		irq_enable(DT_INST_IRQN(inst));			\
+	}
+
+#define STM32_COMP_DEVICE(inst)						\
+	static struct stm32_pclken comp_clk[] = STM32_DT_INST_CLOCKS(inst); \
```

Comment:
`clocks` property seems mandatory. Add `required` in the DT bindings YAML file.

Reply:
added